### DISCUSSION TITLE
220516 연로그 인터뷰 피드백

### DISCUSSION THREAD

**Discussion by YJGwon** [woowacourse-study/2022-lv2-effective-java-interview#9] [General]
### 인터뷰어 -> 인터뷰이

> 점수를 매겨주셔도 좋고, 체크 형식으로 작성해주신 뒤 짧은 코멘트를 남겨주셔도 좋습니다. 해당 항목에 대해 자유롭게 의견을 남겨주세요.

- 말하기 태도, 어조가 적절했나요?
- 전달하고자 하는 바를 명료하게 전달했나요?
- 본인만의 주관이 잘 드러났나요?
- 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- 모르는 부분에 대해 잘 대처했나요?

### 인터뷰이 -> 인터뷰어

> 인터뷰어로써의 역량을 키우려는 것이 아닌 저희 스터디의 원활한 진행을 위한 피드백입니다. 각 크루가 아닌 전반적인 진행 상황에 대해 의견을 남겨주세요!

- 사전에 공유한 내용이 잘 전달되었나요?
- 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- 새로운 관점을 배울 수 있었나요?

**Comment by kyukong** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
- [x] 본인만의 주관이 잘 드러났나요?
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- [x] 모르는 부분에 대해 잘 대처했나요?

Optional 을 써야 하는 상황에 대한 연로그의 주관이 뚜렷하여 질문할 꺼리가 많이 없었네요...!! 인터뷰 처음부터 끝까지 동일한 주관으로 연로그의 생각을 차분하게 설명해주어 좋았습니다. 준비를 정말 많이 한게 느껴졌어요. 사실 모르는 부분이 있었나 싶지만 모든 대답에 잘 대처하여 체크했습니다! 수고하셨습니다~!

**Comment by yeon-06** [comment]
- [x] 사전에 공유한 내용이 잘 전달되었나요?
- [x] 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- [x] 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- [ ] 새로운 관점을 배울 수 있었나요?

발표하다 중간에 말 꼬여서 좀 얼굴 화끈거렸는데 다들 좋게 봐주셔서 감사하네요 ㅎㅎ

**Comment by Ohzzi** [comment]
말하기 태도, 어조가 적절했나요?
- 말 빠르기도 적당하고 발음이 명료해서 좋았습니다!

전달하고자 하는 바를 명료하게 전달했나요?
- 준비가 워낙 잘 되어 있어서 그런지 명쾌하게 전달해주셨어요!

본인만의 주관이 잘 드러났나요?
- JdbcTemplate에서 직접 터뜨리는 예외를 바로 사용하는 것이 아닌 Optional 처리를 해 준 부분에 기준이 확실하게 있었다고 생각합니다.

책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- 미션에서 사용했고, 직접 Optional을 반환하지 않는 JdbcTemplate를 예시로 들면서 책의 내용을 넘어서 실제 활용에 대해 고민한 부분이 좋았어요

모르는 부분에 대해 잘 대처했나요?
- 모르는 부분이... 있었나요...?

**Comment by YJGwon** [comment]
- [X] 말하기 태도, 어조가 적절했나요?
- [X] 전달하고자 하는 바를 명료하게 전달했나요?
- [X] 본인만의 주관이 잘 드러났나요?
- [X] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- [x] 모르는 부분에 대해 잘 대처했나요?

신로그..그는 연인가..?
일단 자료가 책 내용만을 요약한 것이 아닌 다른 관점에서도 이해해주신 자료였던게 인상깊었습니다. 또한 본인만의 경험을 바탕으로 해당 내용을 깊게 고민해보신 점이 좋았습니다. 질문의 의도를 파악하고 기대 이상의 답변을 들려주시네요!
그런데 엄~!~!청 잘 준비해주신거에 비해서 약간 긴장한 듯한 느낌이 있었어요. 자신감을 가져요 신로그!!

**Comment by ulimy** [comment]
자신의 의견을 차분히 들려주셔서 듣기 편했습니다!  중간에 살짝 말이 꼬이신 것 같았는데 크게 당황한 티 나지 않아서 눈치를 못챌뻔 했네요!
풍부한 자료 덕분에 이해하기가 수월했어요 :) 
오늘의 다짐 : 준비 안했다는 연로그 말 믿지 않기ㅎㅎ

**Comment by tco0427** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
  -> 긴장은 안하신 것 같은데..목소리가 살짝 떨리는 느낌을 받았어요. 하지만 이부분을 제외하면 완벽했습니다!
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
  -> 깊이 있게 공부하였다는 것이 느껴질 정도로 본인이 이야기하고자 하는 것이 확실히 전달된 것 같스빈다.
- [x] 본인만의 주관이 잘 드러났나요?
  -> 넵!
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
  -> 네. 직접 성능 측정 하는 부분 등에서 직접 경험해보고, 책의 내용을 수용한다는 것이 느껴졌습니다.
- [x] 모르는 부분에 대해 잘 대처했나요?
  -> 모르는 부분이 없던데요...

**Comment by YJGwon** [comment]
### 질문

- null과 빈 리스트 반환 시의 성능차이 확인하셨는데, Optional 생성에 비용이 많이 든다는 이야기가 많다. 왜 Optional을 선택해야 할까?
- jdbcTemplate으로부터 받은 결과를 Optional로 반환했는지? 그렇게 한 후 Service에서 null check를 해서 예외를 throw할텐데, Dao에서도 할 수 있었는데 굳이 Optional로 반환한 이유?
- Dao에 Optional 적용하기 전에 느꼈던 문제점? 이 것이 어떻게 해결되었다고 느꼈는지?
- Optional을 사용하는 이유가 값이 없는 경우를 예외라고 보지 않고 다르게 처리하기 위함인데, 예외를 발생시켜야 하는 경우라면?
- queryForObject에서 2개 이상의 결과가 나올때는?
    - Dao의 예시(queryForObject)에서, 결과값이 0개일때와 2개일때를 다르게 처리해야한다고 말한 이유?
- JdbcTemplate 내부 설계를 뜯어본 후 느낀 점?
- Dao이외에 앞으로 어떨 때 Optional을 고려할건지?
